js/modules/k6/experimental/fs/file.go

```diff
@@ -0,0 +1,142 @@
+package fs
+
+import (
+	"io"
+	"path/filepath"
+	"sync/atomic"
+
+	"go.k6.io/k6/lib"
+)
+
+// file is an abstraction for interacting with files.
+type file struct {
+	path string
+
+	// data holds a pointer to the file's data
+	data []byte
+
+	// offset holds the current offset in the file
+	//
+	// TODO: using an atomic here does not guarantee ordering of reads and seeks, and leaves
+	// the behavior not strictly defined. This is something we might want to address in the future, and
+	// is tracked as part of #3433.
+	offset atomic.Int64
+}
+
+// Stat returns a FileInfo describing the named file.
+func (f *file) stat() *FileInfo {
+	filename := filepath.Base(f.path)
+	return &FileInfo{Name: filename, Size: len(f.data)}
```

Comment:
nit: this is already impleemented: 
```suggestion
	return &FileInfo{Name: filename, Size: f.size()}
```